src/AWSMSKAuthTokenGenerator.cs

```diff
@@ -1,225 +1,306 @@
+    /// <param name="profileName">AWS Credentials to sign the request will be fetched from this profile</param>
+    /// <param name="region">Region of the MSK cluster</param>
+    /// <returns> A tuple containing Auth token in string format and it's expiry time </returns>
+    public (string, long) GenerateAuthTokenFromProfile(string profileName, RegionEndpoint region)
+    {
+        var chain = new CredentialProfileStoreChain();
+
+        if (chain.TryGetAWSCredentials(profileName, out var awsCredentials))
+        {
+            return GenerateAuthTokenFromCredentialsProvider(() => awsCredentials, region, false).GetAwaiter().GetResult();
         }
 
-        /// <summary>
-        /// Generate a token for IAM authentication to an MSK cluster using an IAM Role
-        /// <remarks>
-        /// This method generates an Auth token using the roleArn provided with the provided SessionName (optional). If SessionName is not provided,
-        /// a default session name of "MSKSASLDefaultSession" is used. Note that this method uses the STS global endpoint to assume role to sign the credentials.
-        /// For more involved use cases like using regional endpoints, consider using the GenerateAuthTokenFromCredentialsProvider method directly. 
-        /// </remarks>
-        /// </summary>
-        /// <param name="region">Region of the MSK cluster</param>
-        /// <param name="roleArn">ARN of the role which needs to be assumed for signing the request</param>
-        /// <param name="sessionName">An optional session name</param>
-        /// 
-        /// <returns> A tuple containing Auth token in string format and it's expiry time </returns>
-        public async Task<(string, long)> GenerateAuthTokenFromRoleAsync(RegionEndpoint region, String roleArn, String sessionName = "MSKSASLDefaultSession")
+        throw new ArgumentException($"Could not find credentials using profile {profileName}");
+    }
+
+    /// <summary>
+    /// Generate a token for IAM authentication to an MSK cluster using an IAM Profile
+    /// <remarks>
+    /// This method generates an Auth token using and IAM Profile
+    /// </remarks>
+    /// </summary>
+    /// <param name="profileName">AWS Credentials to sign the request will be fetched from this profile</param>
+    /// <param name="region">Region of the MSK cluster</param>
+    /// <returns> A tuple containing Auth token in string format and it's expiry time </returns>
+    public Task<(string, long)> GenerateAuthTokenFromProfileAsync(string profileName, RegionEndpoint region)
+    {
+        var chain = new CredentialProfileStoreChain();
+
+        if (chain.TryGetAWSCredentials(profileName, out var awsCredentials))
         {
-            var assumeRoleReq = new AssumeRoleRequest()
-            {
-                RoleSessionName = sessionName,
-                RoleArn = roleArn
-            };
+            return GenerateAuthTokenFromCredentialsProvider(() => awsCredentials, region).AsTask();
```

Comment:
Why change from `await` to `AsTask()` here?

Reply:
Since this is the Async method that returns the Task. I call the method GenerateAuthTokenFromCredentialsProvider() to be executed async and I just return that task. The task will be waited from the client ( that will use the method ).
When that will be waited, even this one will be waited then. It might be something improvement, but not something big, just I escalate the wait one step further.

Reply:
I generally prefer to follow the guidance here and `await`, but agree it shouldn't be a big functional difference https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md#prefer-asyncawait-over-directly-returning-task